pkg/server/httpAPI.go

```diff
@@ -384,7 +387,68 @@ func (h *HTTPHandler) serveLatestSnapshotMetadata(rw http.ResponseWriter, req *h
 	}
 	rw.WriteHeader(http.StatusOK)
 	rw.Write(json)
-	return
+}
+
+func (h *HTTPHandler) serveConfig(rw http.ResponseWriter, req *http.Request) {
+	inputFileName := "/var/etcd/config/etcd.conf.yaml"
+	outputFileName := "/etc/etcd.conf.yaml"
+	configYML, err := ioutil.ReadFile(inputFileName)
+	if err != nil {
+		h.Logger.Warnf("Unable to read etcd config file: %v", err)
+		rw.WriteHeader(http.StatusInternalServerError)
+		return
+	}
+
+	config := map[string]interface{}{}
+	if err := yaml.Unmarshal([]byte(configYML), &config); err != nil {
+		h.Logger.Warnf("Unable to unmarshal etcd config yaml file: %v", err)
+		rw.WriteHeader(http.StatusInternalServerError)
+		return
+	}
+
+	// fetch pod name from env
+	podName := os.Getenv("POD_NAME")
+	config["name"] = podName
+
+	protocol := "http"
+	if h.EnableTLS {
+		protocol = "https"
+	}
+
+	initAdPeerURL := config["initial-advertise-peer-urls"]
+	svcName, namespace, peerPort, err := parsePeerURL(fmt.Sprint(initAdPeerURL))
+	if err != nil {
+		h.Logger.Warnf("Unable to determine service name, namespace, peer port from advertise peer urls : %v", err)
+		rw.WriteHeader(http.StatusInternalServerError)
+		return
+	}
+	domaiName := fmt.Sprintf("%s.%s.%s", svcName, namespace, "svc.cluster.local")
+	config["initial-advertise-peer-urls"] = fmt.Sprintf("%s://%s.%s:%s", protocol, podName, domaiName, peerPort)
+
+	data, err := yaml.Marshal(&config)
+	if err != nil {
+		h.Logger.Warnf("Unable to marshal data to etcd config yaml file: %v", err)
+		rw.WriteHeader(http.StatusInternalServerError)
+		return
+	}
+
+	if err := ioutil.WriteFile(outputFileName, data, 0644); err != nil {
+		h.Logger.Warnf("Unable to write etcd config file: %v", err)
+		rw.WriteHeader(http.StatusInternalServerError)
+		return
+	}
+
+	http.ServeFile(rw, req, outputFileName)
+
+	h.Logger.Info("Served config for ETCD instance.")
+}
+
+func parsePeerURL(peerURL string) (string, string, string, error) {
+	tokens := strings.Split(peerURL, "@")
```

Comment:
I didn’t get why we are splitting at `@`?

Reply:
We are clear about this now, right?